### PR TITLE
refactor(cdk/schematics): change usages of whitelist in selector interfaces to fileTypeFilter

### PR DESCRIPTION
**Pull request by josephperrott**
Migrate usages of whitelist in each of the ng-update selector interfaces to
be fileTypeFilter instead.

### CODE REVIEW

File: src/cdk/schematics/ng-update/data/input-names.ts [hunk 1]
```diff
@@ -14,8 +14,11 @@ export interface InputNameUpgradeData {
   replace: string;
   /** The new name for the @Input(). */
   replaceWith: string;
-  /** Whitelist where this replacement is made. If omitted it is made in all HTML & CSS */
-  whitelist: {
+  /**
+   * Controls which file types in which this replacement is made. If omitted, it is made in all
+   * files.
+   */
+  fileTypeFilter: {
```
Comment (devversion) [inline src/cdk/schematics/ng-update/data/input-names.ts:21]: This doesn't seem about file types for this data set. I can't come up with a good name either, but maybe something around `limitedTo: {attributes: ['a', 'b']}`

This comment also applies to similar data sets where we don't filter by file type. I think `fileTypeFilter` is not 100% accurate either as the options are HTML, stylesheet and strings. Obviously "strings" is a bit different here. Maybe we can change that to something better, or change the option name (as proposed with the comment below)

**Review by devversion — COMMENTED** [review]
I know this is a draft, but I'm providing early feedback in case it helps (due to it being a larger refactor)

File: src/cdk/schematics/ng-update/data/input-names.ts (hunk 1 above)
Comment (josephperrott) [reply src/cdk/schematics/ng-update/data/input-names.ts:21]: Done, I like `limitedTo` as it reads nicely in dot notation.

**Review by jelbourn — APPROVED** [review]
LGTM

File: src/cdk/schematics/ng-update/update-schematic.md [hunk 2]
```diff
@@ -132,7 +132,7 @@ export const propertyNames: VersionChanges<MaterialPropertyNameData> = {
         {
           replace: 'color',
           replaceWith: 'newColor',
-          whitelist: {
+          fileTypeFilter: {
```
Comment (devversion) [inline src/cdk/schematics/ng-update/update-schematic.md:135]:
```suggestion
          limitedTo: {
```

**Pull request merged**